app/views/layouts/angular-bootstrap/_auth_keypair_angular_bootstrap.html.haml

```diff
@@ -0,0 +1,81 @@
+- ng_show                   ||= true
+- validate_url              ||= 'create'
+- prefix                    ||= 'ssh_keypair'
+- userid_label              ||= _("Username")
+- private_key_label         ||= _("Private Key")
+- new_private_key_label     ||= _("New ") + private_key_label
+- ng_reqd_userid            ||= false
+- ng_reqd_private_key       ||= false
+- userid_disabled           ||= false
+- private_key_disabled      ||= false
+- change_stored_private_key ||= _("Change stored private key")
+- cancel_private_key_change ||= _("Cancel private key change")
+- verify_title_off          ||= _("Server information, username and private key fields are needed to perform verification of credentials")
+
+%div{"ng-controller" => "emsKeypairController as vm", "ng-init" => "vm.initialize(#{ng_model}, formId)"}
+  %div{"ng-show" => "#{ng_show}"}
+    .form-group{"ng-class" => "{'has-error': angularForm.#{prefix}_userid.$error.required}"}
+      %label.col-md-2.control-label{"for" => "textInput-markup"}
+        = userid_label
+      .col-md-4
+        %input.form-control{"type"                => "text",
+                            "id"                  => "textInput-markup",
+                            "ng-required"         => "#{ng_reqd_userid}",
+                            "ng-disabled"         => userid_disabled,
+                            "name"                => "#{prefix}_userid",
+                            "ng-model"            => "#{ng_model}.#{prefix}_userid",
+                            "checkchange"         => ""}
+        %span.help-block{"ng-show" => "angularForm.#{prefix}_userid.$error.required"}
+          = _("Required")
+        .note
+          {{note}}
+
+  %div{"ng-show" => "#{ng_show}"}
+    .form-group{"ng-class" => "{'has-error': angularForm.#{prefix}_password.$error.required}"}
+      %label.col-md-2.control-label{"for" => "textInput-markup"}
+        %div{"ng-show" => "!vm.changeKey"}
+          = private_key_label
+        %div{"ng-show" => "vm.changeKey"}
+          = new_private_key_label
+      .col-md-4
+        %input.form-control{"type"                  => "file",
```

Comment:
In order to show bulleted dots in the Edit mode, `type` needs to be changed to  `text` and should go back to `file` if the user clicks on the `Change stored private key` link.
So maybe you could enclose this block in a `ng-if`...something like this -
```
.col-md-4
   %div{"ng-if" => "vm.showVerifyPrivateKey('#{prefix}_userid')"}
      %input.form-control{"type"                  => "file",
                          "id"                    => "textInput-markup",
                          "ng-required"           => "#{ng_reqd_private_key}",
                          "ng-disabled"           => "!vm.showVerifyPrivateKey('#{prefix}_userid')" || private_key_disabled,
                          "name"                  => "#{prefix}_password",
                          "ng-model"              => "#{ng_model}.#{prefix}_password",
                          "prefix"                => "#{prefix}",
                          "checkchange"           => "",
                          "require-file"          => ""}
    %div{"ng-if" => "!vm.showVerifyPrivateKey('#{prefix}_userid')"}
      %input.form-control{"type"                  => "text",
                          "id"                    => "textInput-markup",
                          "ng-required"           => "#{ng_reqd_private_key}",
                          "ng-disabled"           => "!vm.showVerifyPrivateKey('#{prefix}_userid')" || private_key_disabled,
                          "name"                  => "#{prefix}_password",
                          "ng-model"              => "#{ng_model}.#{prefix}_password",
                          "prefix"                => "#{prefix}",
                          "checkchange"           => "",
                          "require-file"          => ""}
```